prior-work/fraunhofer/directory-td.jsonld

```diff
@@ -0,0 +1,98 @@
+{
+    "@context": "https://www.w3.org/2019/wot/td/v1",
+    "title": "LinkSmart Thing Directory",
+    "securityDefinitions": {
+      "bearer_sc": {
+        "scheme": "bearer"
+      },
+      "basic_sc": {
+        "scheme": "basic"
+      }
+    },
+    "security": "bearer_sc",
+    "actions": {
+      "td": {
+        "description": "Add, modify, or remove a Thing Description",
+        "uriVariables": {
+          "id": {
+            "title": "Thing Description ID",
+            "type": "string",
+            "format": "iri-reference"
+          }
+        },
+        "forms": [
+          {
+            "href": "/td",
+            "htv:methodName": "POST",
+            "contentType": "application/ld+json"
+          },
+          {
+            "href": "/td/{id}",
+            "htv:methodName": "PUT",
+            "contentType": "application/ld+json"
+          },
+          {
+            "href": "/td/{id}",
+            "htv:methodName": "PATCH",
+            "contentType": "application/ld+json"
+          },
+          {
+            "href": "/td/{id}",
+            "htv:methodName": "DELETE"
+          }
+        ]
+      }
```

Comment:
This is a bit of a strange example of an action because it's really several different actions, as I think is revealed by the ambiguous name. I would usually expect an action to be describable with a verb or verb phrase.

Reply:
I had my concerns when mapping the [OpenAPI specs](https://github.com/linksmart/thing-directory/blob/master/apidoc/openapi-spec.yml) to TD ([see the last slide](https://github.com/w3c/wot/blob/master/PRESENTATIONS/2020-06-online-f2f/2020-06-22-Discovery-ThingDirectory-Tavakolizadeh.pdf)). 

I agree that they are different actions, and separating them makes it clearer. However, I would keep the names equivalent to the HTTP methods, because they have their own semantics.
```json
{
  ...
  "actions": {
    "postTD": {
      "description": "Add an anonymous Thing Description (without ID)",
      "forms": [
        {
          "href": "/td",
          "htv:methodName": "POST",
          "contentType": "application/ld+json"
          // how to specify that the response contains a system-generated id in Location header?
        }
      ]
    },
    "putTD": {
      "description": "Add or modify a Thing Description",
      "uriVariables": {
        "id": {
          "title": "Thing Description ID",
          "type": "string",
          "format": "iri-reference"
        }
      },
      "forms": [
        {
          "href": "/td/{id}",
          "htv:methodName": "PUT",
          "contentType": "application/ld+json"
        }
      ]
    },
    "patchTD": {
      "description": "Modify parts of a Thing Description",
      "uriVariables": {
        "id": {
          "title": "Thing Description ID",
          "type": "string",
          "format": "iri-reference"
        }
      },
      "forms": [
        {
          "href": "/td/{id}",
          "htv:methodName": "PATCH",
          "contentType": "application/ld+json"
        }
      ]
    },
    "deleteTD": {
      "description": "Remove a Thing Description",
      "uriVariables": {
        "id": {
          "title": "Thing Description ID",
          "type": "string",
          "format": "iri-reference"
        }
      },
      "forms": [
        {
          "href": "/td/{id}",
          "htv:methodName": "DELETE"
        }
      ]
    }
  }
  ...
}
```